server/fleetv2_http_api/impl/controllers.py

```
@@ -84,21 +114,36 @@ def login(device: Optional[str] = None) -> WerkzeugResponse | Response | tuple[d
+            return _log_info_and_respond(auth_json, 200, "Device authentication initialized.")
+        except _OAuthAuthenticationNotSet as e:
+            msg = "Cannot get device authentication. Keycloak authentication is not set on the HTTP API."
+            _log_debug(str(e))
+            return _log_info_and_respond(msg, 500, msg)
         except Exception as e:
             msg = "Problem reaching oAuth service."
             _log_debug(str(e))
             return _log_info_and_respond(msg, 500, msg)
-        return _log_info_and_respond(auth_json, 200, "Device authentication initialized.")
+
     elif device != None:
```

Ruff

Comparison to `None` should be `cond is not None`

Replace with `cond is not None`

(E711)